eslint.config.js

```diff
@@ -274,6 +274,7 @@ module.exports = [
       "no-func-assign": "off",
       "import/no-extraneous-dependencies": "off",
       "import/no-unresolved": "off",
+      "@typescript-eslint/prefer-optional-chain": "off",
```

Comment:
@JoshuaKGoldberg If we set `parserOptions.ecmaVersion` to `5`, will `typescript-eslint` respect the `ecmaVersion` and automatically turn off this rule?

Reply:
Interesting idea! No, rules in the preset configs won't be configured differently based on your `parserOptions`. I don't believe ESLint configs can be dynamic in this way. At least legacy ones aren't, and we haven't added official support for flat configs yet.

Reply:
@JoshuaKGoldberg Thank you! I think this is not a very common use case, since most TS projects will use `tsc` to transpile ts sources anyway.

@liuxingbaoyu Can we set `parserOptions.ecmaVersion` to `5` for helpers? So the linter should throw if there are non-ES5 codes.

Reply:
`Parsing error: ImportDeclaration should appear when the mode is ES6 and in the module context`.
I tried it and it didn't work.
Our current `ts` source actually has some es6 syntax such as `trailingComma` handled by `terser` which I think is fine.
I have now unified `js` with `ts` and explicitly specified `terser` as es5.